crates/anvil/src/config.rs

```diff
@@ -834,7 +834,13 @@ impl NodeConfig {
 
         // if provided use all settings of `genesis.json`
         if let Some(ref genesis) = self.genesis {
-            genesis.apply(&mut env);
+            env.cfg.chain_id = genesis.config.chain_id;
+            env.block.timestamp = U256::from(genesis.timestamp);
+            if let Some(base_fee) = genesis.base_fee_per_gas {
+                env.block.basefee = U256::from(base_fee);
+            }
+            env.block.number = U256::from(genesis.nonce);
+            env.block.coinbase = genesis.coinbase;
```

Comment:
this applied these fields conditionally before (see `genesis.apply()`)—but they're not conditional on alloy-genesis. Should check for any default values or just apply them like this?